api/tests/auth.test.ts

```diff
@@ -1,120 +1,99 @@
+  const response: GenqlError = await signIn(ctx.client).catch((e) => e);
   expect(response.errors[0].message).toBe('User not found');
 });
 
 test('Authentication: sign up with same username', async () => {
-  const signUpMutation = getSignUpMutation();
-  await executeGraphQlQuery(signUpMutation);
-  const response = (await executeGraphQlQuery(signUpMutation)) as any;
+  await signUp(ctx.client);
+  const response: GenqlError = await signUp(ctx.client).catch((e) => e);
   expect(response.errors[0].message).toBe('Username already taken');
 });
 
 test('Authentication: access protected endpoint without signing in', async () => {
-  const response = (await executeGraphQlQuery(meQuery)) as any;
+  const response: GenqlError = await getMe(ctx.client).catch((e) => e);
   expect(response.errors[0].message).toBe('Not authenticated');
 });
 
 test('Authentication: sign up, sign in with wrong password', async () => {
-  await executeGraphQlQuery(getSignUpMutation());
-
-  const singInIncorrectPassword = {
-    variables: {
-      user: {
-        username: USERNAME,
-        password: 'wrong',
-      },
-    },
-    query: signInMutation.query,
-  };
-  const signInResponse = (await executeGraphQlQuery(
-    singInIncorrectPassword,
-  )) as Record<string, any>;
-  expect(signInResponse?.errors[0].message).toBe('Invalid password');
+  await signUp(ctx.client);
+
+  const signInResponse: GenqlError = await signIn(ctx.client, USERNAME, 'wrong').catch((e) => e);
+  expect(signInResponse.errors[0].message).toBe('Invalid password');
 });
 
 test('Authentication: access protected endpoint without valid token', async () => {
-  ctx.client.setHeader('Authorization', 'Bearer heavy');
-  const response = (await executeGraphQlQuery(meQuery)) as any;
+  setHeader({ Authorization: 'Bearer heavy' });
+  const response = await getMe(ctx.client).catch((e) => e);
   expect(response.errors[0].message).toBe('Invalid authentication token');
 });
 
 test('Authentication: token expiration error', async () => {
-  await executeGraphQlQuery(getSignUpMutation());
+  await signUp(ctx.client);
 
   vi.spyOn(env, 'JWT_EXPIRATION_PERIOD', 'get').mockReturnValue('1s');
-  const signInResponse = (await executeGraphQlQuery(signInMutation)) as Record<
-  string,
-  any
-  >;
-  expect(signInResponse?.signIn?.token).toBeTruthy();
-  const expiry = new Date(signInResponse?.signIn?.session?.referenceExpiryDate).getTime();
+  const signInResponse = await signIn(ctx.client);
+  expect(signInResponse.signIn.token).toBeTruthy();
+  const expiry = new Date(signInResponse.signIn.session.referenceExpiryDate!).getTime();
   const now = new Date().getTime();
 
   const token = signInResponse?.signIn?.token;
-  ctx.client.setHeader('Authorization', `Bearer ${token}`);
+  setHeader({ Authorization: `Bearer ${token}` });
 
   // wait until token expires
   // eslint-disable-next-line no-promise-executor-return
   await new Promise((resolve) => setTimeout(resolve, expiry - now));
```

Comment:
this breaks on my local machine every other (every second) test run when the whole suite runs.

"Breaks" means: test times out because this promise waits too long. It waits too long because the `expiry` variable has a high value (although the env var value is mocked) for no visible reason

> (i like vitest)